examples/CarHandling/CarHandlingDemo.cpp

```diff
@@ -0,0 +1,423 @@
+	btDefaultCollisionConfiguration* collisionConfiguration = new btDefaultCollisionConfiguration();
+
+	//Use the default collision dispatcher. For parallel processing you can use a diffent dispatcher (see Extras/BulletMultiThreaded)
+	btCollisionDispatcher* dispatcher = new	btCollisionDispatcher(collisionConfiguration);
+
+	//btDbvtBroadphase is a good general purpose broadphase. You can also try out btAxis3Sweep.
+	btBroadphaseInterface* overlappingPairCache = new btDbvtBroadphase();
+
+	//The default constraint solver. For parallel processing you can use a different solver (see Extras/BulletMultiThreaded)
+	btSequentialImpulseConstraintSolver* solver = new btSequentialImpulseConstraintSolver;
+
+	//Creates the dynamics world, wich will be responsable for managing our physics objects and constraints
+	this->dynamicsWorld = new btDiscreteDynamicsWorld(dispatcher, overlappingPairCache, solver, collisionConfiguration);
+
+	this->dynamicsWorld->setGravity(btVector3(0, -10, 0));
+
+	guiHelper->createPhysicsDebugDrawer(this->dynamicsWorld);
+
+	//keep track of the shapes, we release memory at exit.
+	//make sure to re-use collision shapes among rigid bodies whenever possible!
+	{
+		//Creates the ground shape
+		btCollisionShape* groundShape = new btBoxShape(btVector3(100, 1, 100));
+
+		//Stores on an array for reusing
+		collisionShapes.push_back(groundShape);
+
+		//Creates the ground rigidbody
+		btRigidBody* groundRigidBody = this->createGroundRigidBodyFromShape(groundShape);
+
+		//Adds it to the world
+		this->dynamicsWorld->addRigidBody(groundRigidBody);
+	}
+
+	{
+		//the dimensions for the boxShape are half extents, so 0,5 equals to 
+		btVector3 halfExtends(1, btScalar(0.5), 2);
+
+		//The btBoxShape is centered at the origin
+		btCollisionShape* chassisShape = new btBoxShape(halfExtends);
+
+		//Stores on an array for reusing
+		collisionShapes.push_back(chassisShape);
+
+		//A compound shape is used so we can easily shift the center of gravity of our vehicle to its bottom
+		//This is needed to make our vehicle more stable
+		btCompoundShape* compound = new btCompoundShape();
+
+		collisionShapes.push_back(compound);
+
+		btTransform localTransform;
+		localTransform.setIdentity();
+		localTransform.setOrigin(btVector3(0, 1, 0));
+
+		//The center of gravity of the compound shape is the origin. When we add a rigidbody to the compound shape
+		//it's center of gravity does not change. This way we can add the chassis rigidbody one unit above our center of gravity
+		//keeping it under our chassis, and not in the middle of it
+		compound->addChildShape(localTransform, chassisShape);
+
+		//Creates a rigid body
+		btRigidBody* chassisRigidBody = this->createChassisRigidBodyFromShape(compound);
+
+		//Adds the vehicle chassis to the world
+		this->dynamicsWorld->addRigidBody(chassisRigidBody);
+
+		btVehicleRaycaster* vehicleRayCaster = new btDefaultVehicleRaycaster(this->dynamicsWorld);
```

Comment:
I was looking for a working vehicle demo and found this PR. It works for me with camera fixes, but I wonder if vehicleRaycaster should be deleted manually. I've checked btDefaultVehicleRaycaster and seems it doesn't own raycaster and doesn't delete it. Also minor question: can raycaster be shared between several vehicles?